sdk/python/kubeflow/training/api/training_client_test.py

```diff
@@ -547,6 +561,370 @@ def __init__(self, kind) -> None:
+        {"name": TEST_NAME, "namespace": FAILED},
+        generate_job_with_status(create_job(), namespace=FAILED),
+    ),
+    (
+        "valid flow with restarting job condition",
+        {"name": TEST_NAME, "namespace": RESTARTING},
+        generate_job_with_status(create_job(), namespace=RESTARTING),
+    ),
+    (
+        "valid flow with running job condition",
+        {"name": TEST_NAME, "namespace": RUNNING},
+        generate_job_with_status(create_job(), namespace=RUNNING),
+    ),
+    (
+        "valid flow with created job condition",
+        {"name": TEST_NAME, "namespace": RUNNING},
+        generate_job_with_status(create_job(), namespace=RUNNING),
+    ),
+    (
+        "valid flow with all parameters set",
+        {
+            "name": TEST_NAME,
+            "namespace": TEST_NAME,
+            "job": create_job(),
+            "job_kind": constants.PYTORCHJOB_KIND,
+            "timeout": 120,
+        },
+        generate_job_with_status(create_job()),
+    ),
+    (
+        "invalid flow with default namespace and a Job that doesn't exist",
+        {"name": TEST_NAME, "job_kind": constants.TFJOB_KIND},
+        RuntimeError,
+    ),
+    (
+        "invalid flow incorrect parameter",
+        {"name": TEST_NAME, "test": "example"},
+        TypeError,
+    ),
+    (
+        "invalid flow withincorrect value",
+        {"name": TEST_NAME, "job_kind": "FailJob"},
+        ValueError,
+    ),
+    (
+        "runtime error case",
+        {
+            "name": TEST_NAME,
+            "namespace": "runtime",
+            "job_kind": constants.PYTORCHJOB_KIND,
+        },
+        RuntimeError,
+    ),
+    (
+        "invalid flow with timeout error",
+        {"name": TEST_NAME, "namespace": TIMEOUT},
+        TimeoutError,
+    ),
+    (
+        "invalid flow with runtime error",
+        {"name": TEST_NAME, "namespace": RUNTIME},
+        RuntimeError,
+    ),
+]
+
+
+test_data_is_job_created = [
+    (
+        "valid flow with default namespace and default timeout",
+        {"name": TEST_NAME},
+        False,  # Default created Job condition is Succeded
+    ),
+    (
+        "valid flow with job that is created",
+        {"name": TEST_NAME, "namespace": CREATED},
+        True,
+    ),
+    (
+        "valid flow with all parameters set",
+        {
+            "name": TEST_NAME,
+            "namespace": CREATED,
+            "job": create_job(),
+            "job_kind": constants.PYTORCHJOB_KIND,
+            "timeout": 120,
+        },
+        True,
+    ),
+    (
+        "invalid flow with default namespace and a Job that doesn't exist",
+        {"name": TEST_NAME, "job_kind": constants.TFJOB_KIND},
+        RuntimeError,
+    ),
+    (
+        "invalid flow incorrect parameter",
+        {"name": TEST_NAME, "test": "example"},
+        TypeError,
+    ),
+    (
+        "invalid flow withincorrect value",
+        {"name": TEST_NAME, "job_kind": "FailJob"},
```

Comment:
Can you create `invalid` const that you can use for such tests cases in job_kind ?